app/src/main/java/io/intrepid/contest/customviews/ClearableEditText.java

```diff
@@ -0,0 +1,108 @@
+package io.intrepid.contest.customviews;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.graphics.drawable.DrawableCompat;
+import android.support.v7.widget.AppCompatEditText;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.AttributeSet;
+import android.view.MotionEvent;
+import android.view.View;
+
+import io.intrepid.contest.R;
+
+public class ClearableEditText extends AppCompatEditText implements View.OnTouchListener, View.OnFocusChangeListener, TextWatcher {
```

Comment:
Small change, let's put a link to where you got this code in a comment within this file.